jormungandr/src/settings/start/mod.rs

```diff
@@ -233,7 +236,15 @@ fn generate_network(
             .clone()
             .unwrap_or(vec![])
             .into_iter()
-            .map(Into::into)
+            .filter_map(|tp| {
+                tp.address.to_addresses().ok().map(|addrs| {
+                    addrs.into_iter().map(move |addr| TrustedPeer {
+                        id: tp.id.clone(),
```

Comment:
Umm, do we have provisions elsewhere for the duplicated peer entries?
These are going to overwrite each other in `PeerMap`. To do it properly, there should be a "happy eyeballs" style async connection routine for a single peer, racing up to two connection trains – one going through a shuffled list of IPv4 addresses, the other through IPv6 – but only to the point where one connects on TCP, then the other needs to be discarded and the protocol handshake with subscriptions should proceed on the winning connection.

Provisions may need to be made in network-grpc to enable two-stage connections.

Reply:
OK, maybe happy eyeballs is a little overkill as there is not much interactive difference in how many milliseconds it takes to establish a connection. So shuffle the list of all addresses for a single peer and go through them with TCP connection attempts until one succeeds, that would do it.

Reply:
Even better perhaps, we could let `hyper` deal with DNS addresses because we give it an URL to connect to.

Reply:
How much headache would it be to implement DNSSEC signature validation within Jormungandr and WARN if provided FQDN is not signed? Seems (I checked just three) Rust libraries do not support DNSSEC validation, does `hyper` rely on OS DNS resolver or a Rust library outside of the OS?

Reply:
`hyper` supports custom resolvers, and previous commits on this branch used `trust-dns-resolver` which supports DNSSec. @eugene-babichenko @NicolasDP what do you think of re-integrating it back, either in this form or hooking into `hyper`?

Reply:
> Umm, do we have provisions elsewhere for the duplicated peer entries?

No, and actually this is my main concern about this pull request which I discussed with @NicolasDP. The **correct** approach is testing IPs until we find a reachable one and then just go for that address. But we must do it right after we got the whole list of addresses, because we also need to feed the node data to `poldercast`. We can rework poldercast crate a bit though to make it able to accept DNS instead of bare IPs.

> Even better perhaps, we could let `hyper` deal with DNS addresses because we give it an URL to connect to.

If the whole network stuff including gossiping is done via `hyper` (that's what I can see from the code), I think it's totally fine.

> what do you think of re-integrating it back, either in this form or hooking into `hyper`?

The whole point of the last commit was to remove an additional dependency, but I feel like we will return if want to support DNSSEC. I'm more into hooking that stuff into `hyper`, just because it sounds way more elegant.

_To sum everything up_, my approach would be to allow using DNS just everywhere in the code. Let's say, make it a first-class citizen along with the IPs. But this may require some modification of external dependencies, specifically `poldercast`.